src/library/autodj/autodjprocessor.cpp

```diff
@@ -505,6 +505,17 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
 TrackPointer AutoDJProcessor::getNextTrackFromQueue() {
     // Get the track at the top of the playlist.
     while (true) {
+        // Incase we start off with < minimum tracks
+        bool randomQueueEnabled = (((m_pConfig->getValueString(
+                    ConfigKey("[Auto DJ]", "EnableRandomQueue")).toInt())) == 1);
+        int minAutoDJCrateTracks = m_pConfig->getValueString(
+                    ConfigKey(kConfigKey, "RandomQueueMinimumAllowed")).toInt();
+        int tracksToAdd = minAutoDJCrateTracks - m_pAutoDJTableModel->rowCount();
+        while (randomQueueEnabled && (tracksToAdd > 0)) {
```

Comment:
>This produces still an endless loop if the connected crates has 0 tracks.

The following after the signal emit should work.
* If no tracks were added , no tracks exist in crate we break.
* If after three attempted retrievals , we couldnt get a track (probably a detached HDD ) , we break ?

 if (tracksToAdd == minAutoDJCrateTracks - m_pAutoDJTableModel->rowCount())
            break;